src/Microsoft.Azure.CosmosEventSourcing/Events/AtomicEvent.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.CosmosEventSourcing.Aggregates;
+using Microsoft.Azure.CosmosEventSourcing.Stores;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.CosmosEventSourcing.Events;
+
+/// <summary>
+/// An event that is used to pass between <see cref="IAggregateRoot"/>'s and the <see cref="IEventStore{TEventItem}"/>
+/// <remarks>This is used to provide optimistic concurrency control for batches of <see cref="DomainEvent"/>'s.</remarks>
+/// </summary>
+public record AtomicEvent : DomainEvent
```

Comment:
```suggestion
public record class AtomicEvent : DomainEvent
```

Reply:
A `record` is implicitly a `class`. With C# 10, you can also now have `record struct`. This is just the more explicit form, saying that you do indeed want a `class` for your `record` type. It's just a style preference, won't change anything here.

https://docs.microsoft.com/dotnet/csharp/language-reference/builtin-types/record

Reply:
Ahh okay, happy to add this thanks for the clarification.